frontend/src/components/PhotoPage.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from "react";
+import Modal from "react-modal";
+import PrimaryCard from "@/components/custom-card/PrimaryCard";
+import { faBriefcase, faXmark } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import CustomButton from "@/components/button/Button";
+import VisitService from "@/services/visit.service";
+import UserService from "@/services/user.service";
+import BuildingInTourService from "@/services/buildingInTour.service";
+import BuildingService from "@/services/building.service";
+
+/**
+ * A detail screen for a picture.
+ * @param photo A photo object.
+ */
+export default function PhotoPage({ photo }) {
+  const [modalIsOpen, setIsOpen] = useState(false);
+  const [person, setPerson] = useState("");
+  const [address, setAddress] = useState("");
+  const [title, setTitle] = useState("");
+
+  useEffect(() => {
+    const getData = async () => {
+      if (photo) {
+        if (photo.state === 1) {
+          setTitle("Aankomst");
+        } else if (photo.state === 2) {
+          setTitle("Vertrek");
+        } else if (photo.state === 3) {
+          setTitle("Extra");
+        }
+        const visit = await VisitService.getEntryByUrl(photo.visit);
+        const user = await UserService.getEntryByUrl(visit.user);
+        const buildingInTour = await BuildingInTourService.getEntryByUrl(
+          visit.building_in_tour
+        );
+        const building = await BuildingService.getEntryByUrl(
+          buildingInTour.building
+        );
+        setPerson(`${user.first_name} ${user.last_name} (${user.email})`);
+        setAddress(
+          `${building.address_line_1} ${building.address_line_2} (${building.nickname})`
+        );
+      }
+    };
+    getData().catch();
+  }, [photo]);
+
+  function openModal() {
+    setIsOpen(true);
+  }
+
+  function closeModal() {
+    setIsOpen(false);
+  }
+
+  function formatTime(time) {
+    const date = new Date(time);
+    return date.toLocaleString("en-GB", {
+      day: "2-digit",
+      month: "2-digit",
+      year: "numeric",
+      hour: "2-digit",
+      minute: "2-digit",
+      second: "2-digit",
+    });
+  }
+
+  return (
+    <div>
+      <button onClick={openModal}>Open Modal</button>
```

Comment:
instead of showing a button with text, I would show a smaller clickable version of the image, which then opens the popup. In the student building page i had defined it like this:

`
<Image
              src={photo.image}
              alt={index}
              width={75}
              height={75}
              unoptimized={true}
              className="rounded-md"
            />`

Reply:
You're right. That would be much better. I'll try to add a way so that you can use whatever you want.

Reply:
This will be done in a seperate PR.